<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
        vecs = torch.stack((tensorA, tensorB), 0)
        vectors_obj = build_vectors(tokens, vecs)

        <a id="change">tensorC = torch.tensor([1, 1], dtype=torch.float)</a>
        vectors_obj[&quotb&quot] = tensorC

        vector_path = os.path.join(self.test_dir, &quotvectors.pt&quot)
        torch.save(vectors_obj.to_ivalue(), vector_path)</code></pre><h3>After Change</h3><pre><code class='java'>
            self.assertEqual(loaded_vectors_obj[&quotb&quot], tensorB)
            self.assertEqual(loaded_vectors_obj[&quotnot_in_it&quot], expected_unk_tensor)

        <a id="change">with self.subTest(&quottorchscript&quot):
            vector_path = os.path.join(self.test_dir, &quotvectors_torchscript.pt&quot)
            torch.save(vectors_obj.to_ivalue(), vector_path)
            loaded_vectors_obj = torch.load(vector_path)

            self.assertEqual(loaded_vectors_obj[&quota&quot], tensorA)
            self.assertEqual(loaded_vectors_obj[&quotb&quot], tensorB)
            self.assertEqual(loaded_vectors_obj[&quotnot_in_it&quot], expected_unk_tensor)

    &#47&#47 we separate out these errors because Windows runs into seg faults when propagating
    &#47&#47 exceptions from C++ using pybind11
   </a> @unittest.skipIf(platform.system() == "Windows", "Test is known to fail on Windows.")
    def test_errors_vectors_cpp(self):
        tensorA = torch.tensor([1, 0, 0], dtype=torch.float)
        tensorB = torch.tensor([0, 1, 0], dtype=torch.float)</code></pre>